DOCSUP-7551: Documented nested with multiple nesting and subcolumns of complex types (#25065)
ClickHouse/ClickHouse: docs/en/sql-reference/data-types/nested-data-structures/nested.md

This example declares the `Goals` nested data structure, which contains data about conversions (goals reached). Each row in the ‘visits’ table can correspond to zero or any number of conversions.

> Arbitrary levels of nesting are supported. Columns of nested structures containing arrays are equivalent to multidimensional arrays, so they have limited support (there is no support for storing these columns in tables with the MergeTree engine).

Reviewer: Need to mention, that it's controlled by setting `flatten_nested` (`true` by default). Arbitrary levels of nesting are supported only, when `flatten_nested` is false. Also need to mention, that format of data, inserted to columns of type `Nested` changes, when `flatten_nested` is false (separate arrays vs array of tuples).

Example:

```sql
SET flatten_nested = 0;

CREATE TABLE t_nest(n Nested(a UInt32, b UInt32)) ENGINE = MergeTree ORDER BY tuple();

SHOW CREATE TABLE t_nest;

┌─statement───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ CREATE TABLE default.t_nest
(
    `n.a` Array(UInt32),
    `n.b` Array(UInt32)
)
ENGINE = MergeTree
ORDER BY tuple()
SETTINGS index_granularity = 8192 │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

INSERT INTO t_nest VALUES ([1, 2], [3, 4]);
```

```sql
SET flatten_nested = 0;

CREATE TABLE t_nest(n Nested(a UInt32, b UInt32)) ENGINE = MergeTree ORDER BY tuple();

SHOW CREATE TABLE t_nest;

┌─statement──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ CREATE TABLE default.t_nest
(
    `n` Nested(a UInt32, b UInt32)
)
ENGINE = MergeTree
ORDER BY tuple()
SETTINGS index_granularity = 8192 │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

INSERT INTO t_nest VALUES ([(1, 3), (2, 4)])
```

> Arbitrary levels of nesting are supported, but to use more than one level you need to set the setting [flatten_nested](../../../operations/settings/settings.md#flatten-nested) to `0`. Columns of nested structures containing arrays are equivalent to multidimensional arrays, so they have limited support (there is no support for storing these columns in tables with the MergeTree engine).

Reviewer 2: ```suggestion
When [flatten_nested](../../../operations/settings/settings.md#flatten-nested) is set to `0` (which is not by default), arbitrary levels of nesting are supported. In this case columns of nested structures, containing arrays, are flattened to multidimensional arrays, so they have limited support (there is no support for storing these columns in tables with the [MergeTree](../../engines/table-engines/mergetree-family/mergetree.md) engine).
```
